### PR TITLE
Tutorial: Cells

### PR DESCRIPTION
**Pull request by brentguf**
Hello fellow RedwoodJS 🌲 folks,

I've added a few comments to clarify why I made the changes in this PR. 

Something else I wanted to ask: if I click the 'scaffolding' link below (won't work here, you can find it on the [Cells page](https://redwoodjs.com/tutorial/cells)) which links to [Creating a Post Editor on the Getting Dynamic tutorial page](https://redwoodjs.com/tutorial/getting-dynamic#creating-a-post-editor), it doesn't bring me to this section, just to the top of the page. Is this Issue-worthy?

> The browser should actually show an array with a number or two (assuming you created a blog post with our [scaffolding](/tutorial/getting-dynamic#creating-a-post-editor) from earlier). Neat!

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -683,16 +682,42 @@ export const Success = ({ blogPosts }) => {
 }
 ```
 
-> When generating you can use any case you'd like and Redwood will do the right thing when it comes to naming. These will all create the same filename:
+> **Cell Generator Naming**
+>
+> When generating cells you can use any case you'd like and Redwood will do the right thing when it comes to naming. However, for our use case you will need _some_ kind of indication that you're using more than one blog post. Calling `yarn redwood g cell blogposts` will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`. These will all create the same filename:
```
Comment (brentguf) [inline TUTORIAL.md:687]: I've read the original text (see quote below) a few times, as well as watched the tutorial on YouTube. 

> You will need _some_ kind of indication that you're using more than one word. Calling `yarn redwood g cell blogposts` will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`

Based on that, I rewrote this part. I'm still not convinced that I understood correctly what the original text tried to say, so let me know if I interpreted it wrongly.

File: TUTORIAL.md [hunk 2]
```diff
@@ -683,16 +682,42 @@ export const Success = ({ blogPosts }) => {
...
 >
 >     yarn rw g cell blog_posts
 >     yarn rw g cell blog-posts
 >     yarn rw g cell blogPosts
 >     yarn rw g cell BlogPosts
->
-> You will need _some_ kind of indication that you're using more than one word. Calling `yarn redwood g cell blogposts` will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`
 
-To get you off and running as quickly as possible the generator assumes you've got a root GraphQL query named the same thing as your cell and gives you the minimum query needed to get something out of the database. In this case it called the query `blogPosts` which is not a valid query name for our existing Posts SDL and Service. We'll have to rename that to just `posts` in both the query name and prop named in `Success`:
+To get you off and running as quickly as possible the generator assumes you've got a root GraphQL query named the same thing as your cell and gives you the minimum query needed to get something out of the database. In this case the query is called `blogPosts`:
```
Comment (brentguf) [inline TUTORIAL.md:694]: I had to move back and forth a bit to follow with this part, so added some clarifications (code example + file references).

File: TUTORIAL.md [hunk 3]
```diff
@@ -788,7 +813,7 @@ export const Success = ({ posts }) => {
         <h2>{post.title}</h2>
       </header>
       <p>{post.body}</p>
-      <div>Posted at: {post.createdAt}</div>
+      <div>Posted at: <time datetime={post.createdAt}>{post.createdAt}</time></div>
```
Comment (brentguf) [inline TUTORIAL.md:816]: Semantic HTML

File: TUTORIAL.md [hunk 4]
```diff
@@ -683,16 +682,42 @@ export const Success = ({ blogPosts }) => {
...
+export const Failure = ({ error }) => <div>Error: {error.message}</div>
+
+export const Success = ({ posts }) => {
+  return JSON.stringify(posts)
+}
+```
+
+However, this is not a valid query name for our existing Posts SDL (`src/graphql/posts.sdl.js`) and Service (`src/services/posts/posts.js`). (To see where these files come from, go back to the [Creating a Post Editor section](https://redwoodjs.com/tutorial/getting-dynamic#creating-a-post-editor) in the *Getting Dynamic* part.)
+
+We'll have to rename that to just `posts` in both the query name and prop named in `Success`:
```
Comment (Tobbe) [inline TUTORIAL.md:720]:
```suggestion
We'll have to rename that to just `posts` in both the query name and in the prop name in `Success`:
```

File: TUTORIAL.md (hunk 1 above)
Comment (Tobbe) [reply TUTORIAL.md:687]: No offence, but tbh I think it was better the way it was originally...

What the tutorial is trying to say is that while both you and I know "blogposts" are two words put together, the generator doesn't know that. So you'll have to give it some kind of hint. Either by fully doing the correct capitalization yourself ("BlogPosts"), or at least make the "p" capital ("blogPosts"). Your other options are to separate the two words with a dash or an underscore, and the generator will know to name the cell BlogPostsCell

Comment (brentguf) [reply TUTORIAL.md:687]: No offence taken ;-) I agree. 

What I did was: I very explicitely wrote how I understood it, so someone could correct me and with the idea to change it again after it's clear. So I will change it again.

Thanks.

File: TUTORIAL.md (hunk 3 above)
Comment (cannikin) [reply TUTORIAL.md:816]: I purposely left out the `<time>` tag here to keep the code simpler and just focus on getting the post data displayed. In part 2 of the tutorial we add Comments and at that point I talk about the `<time>` tag and even add a test for its presence!

File: TUTORIAL.md (hunk 1 above)
Comment (cannikin) [reply TUTORIAL.md:687]: Hmmm...I'm not loving this restructure. I like the flow better of showing users the examples of possible syntax first and then giving the counter example (no case change). 

I'd probably just add a clarification around what "more than one word" means after your note about how you were confused. Something like:

> You will need _some_ kind of indication that you're using more than one word ("blog" and "posts"). Just calling `yarn redwood g cell blogposts` would generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js` which, in this case, is not what we want.

**Review by cannikin — CHANGES_REQUESTED** [review]

File: TUTORIAL.md (hunk 3 above)
Comment (brentguf) [reply TUTORIAL.md]: Great, removed it again!

File: TUTORIAL.md (hunk 1 above)
Comment (brentguf) [reply TUTORIAL.md]: @cannikin I kept the structure now and added some clarifications. I didn't notice that was a counter example, so I also clarified that.

Let me know if you can live with my suggested changes. :)

Comment (cannikin) [reply TUTORIAL.md]: > Calling `yarn redwood g cell blogposts` will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`. These will all create the same filename:

This all ran together for me and it sounded like the command `yarn redwood g cell blogposts` (the command we're demonstrating as the "wrong" command) would somehow create the four variants listed right after...I feel like there should at least be a paragraph break before the "These will all create..." sentance.

However, I still prefer the original way: show the correct command(s) first (using camelCase or snake-case) and then an explanation of why we're doing it that way, with the counter-example of what happens when you try just "blogposts".

**Review by cannikin — CHANGES_REQUESTED** [review]

File: TUTORIAL.md (hunk 1 above)
Comment (brentguf) [reply TUTORIAL.md]: @cannikin Thanks for some more clarifications! Are you sure that you checked the latest version of the file? This is what it looks like now:

> **Indicating Multiplicity to the Cell Generator**
>
> When generating a cell you can use any case you'd like and Redwood will do the right thing when it comes to naming. These will all create the same filename (`web/src/components/BlogPostsCell/BlogPostsCell.js`):
>
>     yarn rw g cell blog_posts
>     yarn rw g cell blog-posts
>     yarn rw g cell blogPosts
>     yarn rw g cell BlogPosts
>
> You will need _some_ kind of indication that you're using more than one word: either snake_case (`blog_posts`), kebab-case (`blog-posts`), camelCase (`blogPosts`) or PascalCase (`BlogPosts`). 
> 
> Calling `yarn redwood g cell blogposts` (without any indication that we're using two words) will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`.

Let me know if there's still anything you'd like to change!

**Comment by brentguf** [comment]
@cannikin Thanks for some more clarifications! Are you sure that you checked the latest version of the file? This is what it looks like now:

> **Indicating Multiplicity to the Cell Generator**
>
> When generating a cell you can use any case you'd like and Redwood will do the right thing when it comes to naming. These will all create the same filename (`web/src/components/BlogPostsCell/BlogPostsCell.js`):
>
>     yarn rw g cell blog_posts
>     yarn rw g cell blog-posts
>     yarn rw g cell blogPosts
>     yarn rw g cell BlogPosts
>
> You will need _some_ kind of indication that you're using more than one word: either snake_case (`blog_posts`), kebab-case (`blog-posts`), camelCase (`blogPosts`) or PascalCase (`BlogPosts`). 
> 
> Calling `yarn redwood g cell blogposts` (without any indication that we're using two words) will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`.

Let me know if there's still anything you'd like to change!

**Comment by brentguf** [comment]
@cannikin Thanks for the clarifications, made some more changes! Are you sure that you were checking the latest version of the file before? This is what it looks like now:

> **Indicating Multiplicity to the Cell Generator**
>
> When generating a cell you can use any case you'd like and Redwood will do the right thing when it comes to naming. These will all create the same filename (`web/src/components/BlogPostsCell/BlogPostsCell.js`):
>
>     yarn rw g cell blog_posts
>     yarn rw g cell blog-posts
>     yarn rw g cell blogPosts
>     yarn rw g cell BlogPosts
>
> You will need _some_ kind of indication that you're using more than one word: either snake_case (`blog_posts`), kebab-case (`blog-posts`), camelCase (`blogPosts`) or PascalCase (`BlogPosts`). 
> 
> Calling `yarn redwood g cell blogposts` (without any indication that we're using two words) will generate a file at `web/src/components/BlogpostsCell/BlogpostsCell.js`.

Let me know if there's still anything you'd like to change!

**Comment by cannikin** [comment]
That sounds great, weird that it doesn't show like that in the diff for me?

image

**Comment by cannikin** [comment]
Hmmmm went back over to the Files tab manually and now it does! I have no idea what GitHub is doing sometimes. :(

**Pull request merged**